portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,42 +14,90 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
 import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Date;
+import java.text.SimpleDateFormat;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns messages/ comments. */
+/** Servlet respondsable for storing and returning messages/comments. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> messages;
-
+  
   @Override
-  public void init() {
-    messages = new ArrayList<>();
-    messages.add("Green tea was founded in China in 2737 B.C.");
-    messages.add("English Breakfast Tea came into existence in 1843");
-    messages.add("Chai tea was invented between 5000 and 9000 years ago");
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = getInput(request, "name-input", "");
+    String message = getInput(request, "message-input", "");
+    long timestamp = System.currentTimeMillis();
+
+    // Convert timestamp into human-readable date
+    SimpleDateFormat formatter= new SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss z");
+    String date = formatter.format(timestamp);
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("name", name);
+    commentEntity.setProperty("message", message);
+    commentEntity.setProperty("date", date);
```

Comment:
I wonder if you can do this timestamp -> date conversion later. It looks like you don't need both date and timestamp in the data store.

Reply:
I just tried it and I was able to do the timestamp -> date conversion in the doGet section! I'll make final touches to that now and commit the change. Thanks for the tip!